test/e2e/specs/site-editor/template-part.spec.js

```diff
@@ -0,0 +1,303 @@
+
+		// Insert a new template block and choose an existing header pattern.
+		await editor.insertBlock( { name: 'core/template-part' } );
+		await editor.canvas.click( 'role=button[name="Choose"i]' );
+		await page.click(
+			'role=listbox[name="Block Patterns"i] >> role=option[name="header"i]'
+		);
+
+		// There are now two header template parts.
+		await expect( headerTemplateParts ).toHaveCount( 2 );
+	} );
+
+	test( 'can convert a single block to a template part', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const paragraphText = 'Test 2';
+
+		await admin.visitSiteEditor();
+
+		// Add a block and select it.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: {
+				content: paragraphText,
+			},
+		} );
+		const paragraphBlock = editor.canvas.locator(
+			`p >> text="${ paragraphText }"`
+		);
+		await editor.selectBlocks( paragraphBlock );
+
+		// Convert block to a template part.
+		await editor.clickBlockOptionsMenuItem( 'Make template part' );
+		await page.type( 'role=dialog >> role=textbox[name="Name"i]', 'Test' );
+		await page.keyboard.press( 'Enter' );
+
+		await page.waitForSelector(
+			'role=button >> text="Template part created."'
+		);
+
+		// Check that the header contains the paragraph added earlier.
+		const templatePartWithParagraph = editor.canvas.locator(
+			'[data-type="core/template-part"]',
+			{ has: paragraphBlock }
+		);
+
+		await expect( templatePartWithParagraph ).toBeVisible();
+	} );
+
+	test( 'can convert multiple blocks to a template part', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const paragraphText1 = 'Test 3';
+		const paragraphText2 = 'Test 4';
+
+		await admin.visitSiteEditor();
+
+		// Add a block and select it.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: {
+				content: paragraphText1,
+			},
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: {
+				content: paragraphText2,
+			},
+		} );
+		const paragraphBlock1 = editor.canvas.locator(
+			`p >> text="${ paragraphText1 }"`
+		);
+		const paragraphBlock2 = editor.canvas.locator(
+			`p >> text="${ paragraphText2 }"`
+		);
+
+		await editor.selectBlocks( paragraphBlock1, paragraphBlock2 );
+
+		// Convert block to a template part.
+		await editor.clickBlockOptionsMenuItem( 'Make template part' );
+		await page.type( 'role=dialog >> role=textbox[name="Name"i]', 'Test' );
+		await page.keyboard.press( 'Enter' );
+
+		await page.waitForSelector(
+			'role=button >> text="Template part created."'
+		);
+
+		// Check that the header contains the paragraph added earlier.
+		const templatePartWithParagraph1 = editor.canvas.locator(
+			'[data-type="core/template-part"]',
+			{ has: paragraphBlock1 }
+		);
+		const templatePartWithParagraph2 = editor.canvas.locator(
+			'[data-type="core/template-part"]',
+			{ has: paragraphBlock2 }
+		);
+
+		// TODO: I couldn't find an easy way to assert that the same template
```

Comment:
I think this is good enough. We probably can do better but I don't think that matters that much 😛.